### ISSUE TITLE
sync bn

### ISSUE THREAD

**Issue by nnop** [nnop/notes#204]
# sync bn

- 李沐 [跨卡同步 Batch Normalization](https://zhuanlan.zhihu.com/p/40496177)
  > 现有的标准 Batch Normalization 因为使用数据并行（Data Parallel），是单卡的实现模式，只对单个卡上对样本进行归一化，相当于减小了批量大小（batch-size）（详见[BN工作原理部分](https://link.zhihu.com/?target=https%3A//zh.mxnet.io/blog/syncbn%23batch-normalization%25E5%25A6%2582%25E4%25BD%2595%25E5%25B7%25A5%25E4%25BD%259C)）。 对于比较消耗显存的训练任务时，往往单卡上的相对批量过小，影响模型的收敛效果。 之前在我们在图像语义分割的实验中，[Jerry](https://link.zhihu.com/?target=http%3A//zhongyuezhang.com/)和我就发现使用大模型的效果反而变差，实际上就是BN在作怪。 跨卡同步 Batch Normalization 可以使用全局的样本进行归一化，这样相当于‘增大‘了批量大小，**这样训练效果不再受到使用 GPU 数量的影响**。 最近在图像分割、物体检测的论文中，使用跨卡BN也会显著地提高实验效果，所以**跨卡 BN 已然成为竞赛刷分、发论文的必备神器**。
- [Implementing Synchronized Multi-GPU Batch Normalization - Zhang Hang](http://hangzh.com/PyTorch-Encoding/notes/syncbn.html)
- [Gluon实现跨卡同步Batch Normlization](https://discuss.gluon.ai/t/topic/1156)
- [batch normalization的multi-GPU版本该怎么实现？](https://www.zhihu.com/question/59321480)
- Synchronize Batch Norm across Multi GPUs #2584 @ pytorch/pytorch
- Synchronized BatchNorm statistics across GPUs #18222 @tensorflow/tensorflow

related papers:

  - MegDet: A Large Mini-Batch Object Detector http://cn.arxiv.org/abs/1711.07240
  - [Path Aggregation Network for Instance Segmentation - cvpr18](http://openaccess.thecvf.com/content_cvpr_2018/papers/Liu_Path_Aggregation_Network_CVPR_2018_paper.pdf)
  - Group Normalization https://arxiv.org/pdf/1803.08494
